### PR TITLE
[JENKINS-58670] Pod Labels UI

### PR DESCRIPTION
**Pull request by cronik**
[JENKINS-58670](https://issues.jenkins-ci.org/browse/JENKINS-58670)

For clusters that run multiple jenkins instance in the same namespace
and also want to limit the number of slave pods by instance, being
able to manage the pod labels is important.

Because there was no UI and thus no DataBoundSetter any labels
set via an init.groovy.d script or script console would get wiped out
on any config change.

This changeset introduces a `labelsStr` property which is used to
bind the data from the UI form. The string format is a simple key=value
pair separated by whitespace. Do to the restricted syntax and character
set defined by kubernetes parsing of the labels is pretty straightforward.

### CODE REVIEW

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud/help-labelsStr.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    </p>
+    <p>
+        Valid label values must be 63 characters or less and must be empty or begin and end with an alphanumeric character
+        (<code>[a-z0-9A-Z]</code>) with dashes (<code>-</code>), underscores (<code>_</code>), dots (<code>.</code>),
+        and alphanumerics between.
+    </p>
+    <p>Examples:
+        <ul>
+            <li><code>jenkins=slave</code></li>
+            <li><code>app.kubernetes.io/managed-by=jenkins app.kubernetes.io/instance=dev</code></li>
```
Comment (Vlatombe) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud/help-labelsStr.html]: According to https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/, `app.kubernetes.io/instance` is a unique name. `app.kubernetes.io/part-of` seems more appropriate.

Comment (cronik) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud/help-labelsStr.html]: right on, I'll refactor to follow that pattern.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java [hunk 2]
```diff
@@ -116,7 +118,7 @@
     private int retentionTimeout = DEFAULT_RETENTION_TIMEOUT_MINUTES;
     private int connectTimeout;
     private int readTimeout;
-    private Map<String, String> labels;
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java]: To achieve backward compatibility, you need to keep the old field as `@deprecated / transient`, then migrate the data in `readResolve()` method.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java [hunk 3]
```diff
@@ -118,8 +118,8 @@
     private int readTimeout;
     /** @deprecated Stored as a list of PodLabels */
     @Deprecated
-    private Map<String, String> labels;
-    private List<PodLabel> podLabels = new ArrayList<PodLabel>();
+    private transient Map<String, String> labels;
```
Comment (cronik) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java]: @Vlatombe Thanks for catching that, I had it in there and removed it when I was testing.

**Pull request merged**